articles/_posts/2015-05-06-io-js-v2.0-released.md

```diff
@@ -0,0 +1,998 @@
+    at Module._compile (module.js:411:25)
+    at Object.Module._extensions..js (module.js:446:10)
+    at Module.load (module.js:353:32)
+    at Function.Module._load (module.js:308:12)
+    at Function.Module.runMain (module.js:469:10)
+    at startup (node.js:124:18)
+    at node.js:882:3
+~~~
+
+<!--
+幅広い環境のサポート (v1.4.0, v1.6.0, v1.8.0 )
+
+今回、かなりビルド環境に気合が入っていて、幅広い環境でバイナリが提供されています。 Node.js の時は、 darwin(osx), linux, sunos(solaris), windows のバイナリが提供されるだけでしたが(それでもかなり素晴らしいけど)、 io.js からは ARM v6, v7も含めたかなり多くの環境でバイナリが提供されています。
+
+binaryは提供されていませんが、 Android でも動くためのビルドオプションがあったり、SmartOS、FreeBSDでも動くためのオプションが提供されています。
+
+また色んなOSの仮想環境を構築し、そこの上でテストを実行しています。どこかの環境でエラーがあったらすぐに分かるようになっています。
+
+https://jenkins-iojs.nodesource.com/job/iojs+any-pr+multi/
+依存ライブラリ関連 (v2.0.0〜)
+
+v8が 4.2 になった
+
+v8が新しくなったので、JavaScriptの最適化やES6の機能が増えました。これに関しては先ほど紹介しました。
+また、v8がアップグレードされたことで、V8のAPI が更新され、 Native Module も変更が必要になりました。
+つまり、古いNative moduleのままでは動かなくなってしまいました。
+
+現時点で v2.0 では動かないモジュール一覧に関しては以下のissueをご一読ください。
+
+github.com
+
+nan の更新を行うだけでいけるモジュールも多いので、忘れているモジュールに関しては、 今のところ nan を更新して pull request で教えてあげればいいかと思います。
+ただ毎回 v8 が更新されるだけで動かなくなるモジュールがたくさんあると困るので、 nan を core に入れる提案もされています。
+
+OpenSSL v1.0.2a になった
+
+OpenSSL が v1.0.2a になり、 crypto の性能が改善されました。
+この辺りの記事が詳しいです。*1
+
+github.com
+http-parser v2.5.0 になった
+
+http-parserがやっと最新になりました。一旦 http-parser は v2.4になった後、バグが見つかりrevertされましたが、そのバグが修正され、高速化されたhttp-parserが使えるようになりました。
+
+libuv が v1.4.2 になった
+
+libuv が新しくなりました。 windows/linuxでのバグ修正や ARM でも動くようにするための修正などが含まれています。
+
+npm が v2.9.0 になった
+
+npm も新しくなりました。翻訳があるのでそこを見ておくと良いかもしれません。
+medium.com
+-->
+
+폭넓은 환경의 지원(v1.4.0, v1.6.0, v1.8.0)
+----
+이번에, 상당히 빌드 환경에 기합이 들어가있어, 폭넓은 환경에서 바이너리가 제공되고 있습니다.
+Node.js였을 때는, darwin(osx), linux, sunos(solaris), windows의 바이너리가 제공되는 것 뿐이였는데(그것만으로도 상당히 대단하긴했지만), io.js부터는 ARM v6, v7을 포함해 상당히 많은 환경에서 바이너리가 제공되고 있습니다.
+
+binary는 제공되지 않지만, Android에서 기동하기위한 옵션이 있거나, SmartOS, FreeBSD에서 기동하기 위한 옵션이 제공되고 있습니다.
+
+또 여러 OS의 가상 환경을 구축하고 그 위에서 테스트를 실행하고 있습니다. 어딘가의 환경에서 에러가 발생하면 즉시 알수 있게 되어있습니다.
+
+[https://jenkins-iojs.nodesource.com/job/iojs+any-pr+multi/](https://jenkins-iojs.nodesource.com/job/iojs+any-pr+multi/)
+
+의존 라이브러리 관련(v2.0.0~)
+----
+v8이 4.2가 되었다.
+----
+v8이 새로워졌기 때문에, JavaScript의 최적화 및 ES6의 기능이 늘어났습니다. 이에 관해서는 좀 전에 소개했습니다. 또 v8이 업그레이드되면서, V8의 API가 갱신되고 Native Module도 변경이 필요하게 됐습니다.
+
+즉, 낡은 Native module의 상태로는 작동하지 않게 되었습니다.
+
+현 시점에서 v2.0에서 동작하지 않는 모듈목록에 관해서는 아래의 issue를 읽어 주세요.
+
+[github.com](https://github.com/iojs/io.js/issues/1620)
+
+[nan](https://github.com/iojs/nan)을 [갱신하는 것만으로](https://github.com/websockets/bufferutil/pull/3) 가능한 모듈도 많기 때문에, 잊고있는 모듈에 관해서는, 지금 nan을 갱신해서 pull request을 통해 알려주는 것이 좋지 않을까 하고 생각합니다.
+다만 이번 v8이 갱신되는 것 만으로, 동작하지 않게 되는 모듈이 많으면 곤란하기 때문에, nan을 core에 넣자는 제안도 있습니다.
+
+OpenSSL v1.0.2a가 되었다.
+----
+OpenSSL이 v1.0.2a가 되고, crypto의 성능이 개선되었습니다.
+이 글이 자세히 설명하고 있습니다. *1
+
+(github.com)[https://github.com/iojs/io.js/wiki/Crypto-Performance-Notes-for-OpenSSL-1.0.2a-on-iojs-v1.8.0]
```

Comment:
`[github.com](https://github.com/iojs/io.js/wiki/Crypto-Performance-Notes-for-OpenSSL-1.0.2a-on-iojs-v1.8.0)`